tests/test_wordpress_api.py

```diff
@@ -27,245 +20,182 @@ def raise_for_status(*args):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test", tags_exclude=[1234, 5678])
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/"
-            + "wp-json/wp/v2/posts?slug=test&tags=&tags_exclude=1234,5678"
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_including_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(slug="test", tags=[1234, 5678])
-        get.assert_called_once_with(
-            "https://admin.insights.ubuntu.com/"
-            + "wp-json/wp/v2/posts?slug=test&tags=1234,5678"
-            + "&tags_exclude="
-        )
-        self.assertEqual(article, "hello_test")
-
-    @patch("canonicalwebteam.http.CachedSession.get")
-    def test_including_and_excluding_article(self, get):
-
-        get.return_value = MockResponse()
-
-        article = api.get_article(
-            slug="test", tags=[1234, 5678], tags_exclude=[9876]
+    @vcr.use_cassette("fixtures/vcr_cassettes/articles_with_metadata.yaml")
+    def test_get_articles_with_metadata(self):
+        # This function is similar to get_articles, plus the metadata
+        # therefore all other functionality is tested via test_get_articles
+        articles, metadata = api.get_articles_with_metadata()
+        self.assertIsNotNone(metadata["total_pages"])
+        self.assertIsNotNone(metadata["total_posts"])
+        self.assertEqual(len(articles), 12)
+
+    @vcr.use_cassette("fixtures/vcr_cassettes/articles.yaml")
+    def test_get_articles(self):
+        articles, total_pages = api.get_articles()
+        self.assertIsInstance(total_pages, str)
+        self.assertEqual(len(articles), 12)
+
+        first_page_articles, _ = api.get_articles(page=1, per_page=6)
+        second_page_articles, _ = api.get_articles(page=2, per_page=6)
+        first_and_second, _ = api.get_articles(per_page=12)
+        self.assertEqual(len(first_page_articles), 6)
+        self.assertEqual(len(second_page_articles), 6)
+        self.assertEqual(len(first_and_second), 12)
+        for article in first_page_articles:
+            self.assertTrue(article in first_and_second)
+        for article in second_page_articles:
+            self.assertTrue(article in first_and_second)
+        for article in second_page_articles:
+            self.assertTrue(article not in first_page_articles)
+
+        articles_without_tags, _ = api.get_articles(tags_exclude=[3278])
+
+        for article in articles_without_tags:
+            self.assertTrue(3278 not in article["tags"])
+
+        articles_all_have_one_tag, _ = api.get_articles(tags=[3278])
+
+        for article in articles_all_have_one_tag:
+            self.assertTrue(3278 in article["tags"])
+
+        # TODO:
+        # https://github.com/canonical-web-and-design/canonicalwebteam.blog/issues/67
```

Comment:
this seems more something that you will be able to add directly when working on it.